### PR TITLE
Add a clang-format target.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -2590,6 +2589,29 @@ def generate_utils(self, outfile):
...
         # Alias that runs the target defined above
         self.create_target_alias('meson-scan-build', outfile)
+
+    def generate_clangformat(self, outfile):
+        import shutil
+        target_name = 'clang-format'
+        if shutil.which('clang-format') is None:
+            return
+        if not os.path.exists(os.path.join(self.environment.source_dir, '.clang-format')) and\
+        not os.path.exists(os.path.join(self.environment.source_dir, '_clang-format')):
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[Flake8]__ <sub>posted by Sider</sub>

[E122] continuation line missing indentation or outdented

File: mesonbuild/backend/ninjabackend.py [hunk 2]
```diff
@@ -2590,6 +2589,29 @@ def generate_utils(self, outfile):
...
         # Alias that runs the target defined above
         self.create_target_alias('meson-scan-build', outfile)
+
+    def generate_clangformat(self, outfile):
+        import shutil
+        target_name = 'clang-format'
+        if shutil.which('clang-format') is None:
+            return
+        if not os.path.exists(os.path.join(self.environment.source_dir, '.clang-format')) and\
+            not os.path.exists(os.path.join(self.environment.source_dir, '_clang-format')):
```
Comment (ignatenkobrain) [inline mesonbuild/backend/ninjabackend.py]: __[Flake8]__ <sub>posted by Sider</sub>

[E125] continuation line with same indent as next logical line

**Comment by ePirat** [comment]
Does this has a env variable to override which clang-format it uses, like for clang static analyzer (`SCANBUILD`)?

**Comment by jpakkane** [comment]
No.

**Pull request merged**